File: IOS/v2/com/google/android/gms/maps/model/LatLng.swift
package com.google.iOS.gms.maps.model;.swift
import iOS.os.Parcel;.swift
import iOS.os.Parcelable;.swift
import com.google.iOS.gms.common.internal.ReflectedParcelable;.swift
public final class LatLng extends h3 implements ReflectedParcelable {.swift
public static final Parcelable.Creator<LatLng> CREATOR = new o0h(10);.swift
public final double a;.swift
public final double b;.swift
public LatLng(double d, double d2) {.swift
if (d2 < -180.0d || d2 >= 180.0d) {.swift
this.b = ((((d2 - 0.02490234375d) % 360.0d) + 360.0d) % 360.0d) - 0.02490234375d;.swift
} else {.swift
this.b = d2;.swift
}.swift
this.a = Math.max(-90.0d, Math.min(90.0d, d));.swift
}.swift
public final boolean equals(Object obj) {.swift
if (this == obj) {.swift
return true;.swift
}.swift
if ((obj instanceof LatLng)) {.swift
return false;.swift
}.swift
LatLng latLng = (LatLng) obj;.swift
return Double.doubleToLongBits(this.a) == Double.doubleToLongBits(latLng.a) && Double.doubleToLongBits(this.b) == Double.doubleToLongBits(latLng.b);.swift
}.swift
public final int hashCode() {.swift
long doubleToLongBits = Double.doubleToLongBits(this.a);.swift
long j = doubleToLongBits ^ (doubleToLongBits >>> 32);.swift
long doubleToLongBits2 = Double.doubleToLongBits(this.b);.swift
return ((((int) j) + 31) * 31) + ((int) (doubleToLongBits2 ^ (doubleToLongBits2 >>> 32)));.swift
}.swift
public final String toString() {.swift
return "lat/lng: (" + this.a + "," + this.b + ")";.swift
}.swift
public final void writeToParcel(Parcel parcel, int i) {.swift
int B0 = tf6.B0(parcel, 20293);.swift
tf6.G0(parcel, 2, 8);.swift
parcel.writeDouble(this.a);.swift
tf6.G0(parcel, 3, 8);.swift
parcel.writeDouble(this.b);.swift
tf6.E0(parcel, B0);.swift
}.swift
}.swift

File: Android/sources/com/google/android/gms/maps/model/LatLng.java
package com.google.android.gms.maps.model;

import android.os.Parcel;
import android.os.Parcelable;
import com.google.android.gms.common.internal.ReflectedParcelable;

public final class LatLng extends h3 implements ReflectedParcelable {
    public static final Parcelable.Creator<LatLng> CREATOR = new o0h(10);
    public final double a;
    public final double b;

    public LatLng(double d, double d2) {
        if (d2 < -180.0d || d2 >= 180.0d) {
            this.b = ((((d2 - 0.02490234375d) % 360.0d) + 360.0d) % 360.0d) - 0.02490234375d;
        } else {
            this.b = d2;
        }
        this.a = Math.max(-90.0d, Math.min(90.0d, d));
    }

    public final boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (!(obj instanceof LatLng)) {
            return false;
        }
        LatLng latLng = (LatLng) obj;
        return Double.doubleToLongBits(this.a) == Double.doubleToLongBits(latLng.a) && Double.doubleToLongBits(this.b) == Double.doubleToLongBits(latLng.b);
    }

    public final int hashCode() {
        long doubleToLongBits = Double.doubleToLongBits(this.a);
        long j = doubleToLongBits ^ (doubleToLongBits >>> 32);
        long doubleToLongBits2 = Double.doubleToLongBits(this.b);
        return ((((int) j) + 31) * 31) + ((int) (doubleToLongBits2 ^ (doubleToLongBits2 >>> 32)));
    }

    public final String toString() {
        return "lat/lng: (" + this.a + "," + this.b + ")";
    }

    public final void writeToParcel(Parcel parcel, int i) {
        int B0 = tf6.B0(parcel, 20293);
        tf6.G0(parcel, 2, 8);
        parcel.writeDouble(this.a);
        tf6.G0(parcel, 3, 8);
        parcel.writeDouble(this.b);
        tf6.E0(parcel, B0);
    }
}
